localstack/services/apigateway/integration.py

```diff
@@ -123,6 +136,97 @@ def invoke(cls, invocation_context: ApiInvocationContext) -> Response:
+        stage = invocation_context.stage
+        headers = invocation_context.headers
+        resource_path = invocation_context.resource_path
+        invocation_context.context = get_event_request_context(invocation_context)
+        try:
+            path_params = extract_path_params(path=relative_path, extracted_path=resource_path)
+            invocation_context.path_params = path_params
+        except Exception:
+            path_params = {}
+
+        func_arn = uri
+        if ":lambda:path" in uri:
+            func_arn = uri.split(":lambda:path")[1].split("functions/")[1].split("/invocations")[0]
+
+        if invocation_context.authorizer_type:
+            authorizer_context = {
+                invocation_context.authorizer_type: invocation_context.auth_context
+            }
+            invocation_context.context["authorizer"] = authorizer_context
+
+        payload = self.request_templates.render(invocation_context)
+
+        # TODO: change this signature to InvocationContext as well!
+        result = lambda_api.process_apigateway_invocation(
+            func_arn,
+            relative_path,
+            payload,
+            stage,
+            api_id,
+            headers,
+            is_base64_encoded=invocation_context.is_data_base64_encoded,
+            path_params=path_params,
+            query_string_params=query_string_params,
+            method=invocation_context.method,
+            resource_path=resource_path,
+            request_context=invocation_context.context,
+            stage_variables=invocation_context.stage_variables,
+        )
+
+        if isinstance(result, FlaskResponse):
+            response = flask_to_requests_response(result)
+        elif isinstance(result, Response):
+            response = result
+        else:
+            response = LambdaResponse()
+            parsed_result = result if isinstance(result, dict) else json.loads(str(result or "{}"))
+            parsed_result = common.json_safe(parsed_result)
+            parsed_result = {} if parsed_result is None else parsed_result
+            response.status_code = int(parsed_result.get("statusCode", 200))
+            parsed_headers = parsed_result.get("headers", {})
+            if parsed_headers is not None:
+                response.headers.update(parsed_headers)
+            try:
+                result_body = parsed_result.get("body")
+                if isinstance(result_body, dict):
+                    response._content = json.dumps(result_body)
+                else:
+                    body_bytes = to_bytes(to_str(result_body or ""))
+                    if parsed_result.get("isBase64Encoded", False):
+                        body_bytes = base64.b64decode(body_bytes)
+                    response._content = body_bytes
+            except Exception as e:
+                LOG.warning("Couldn't set Lambda response content: %s", e)
+                response._content = "{}"
+            self.update_content_length(response)
```

Comment:
nit: we could remove this line here, and instead move it to line 226 (replacing the `...headers["Content-Length"] = ...` assignment there)